persistence/javadsl/src/main/scala/com/lightbend/lagom/javadsl/persistence/PersistentEntity.scala

```diff
@@ -191,19 +191,17 @@ abstract class PersistentEntity[Command, Event, State] {
      * Create a `BehaviorBuilder` that corresponds to this `Behavior`, i.e. the builder
      * is populated with same state, same event and command handler functions.
      */
-    def builder(): BehaviorBuilder = new BehaviorBuilder(state, eventHandlers, commandHandlers)
+    def builder(): BehaviorBuilder = new BehaviorBuilderImpl(state, eventHandlers, commandHandlers)
 
   }
 
-  /**
-   * Mutable builder that is used for defining the event and command handlers.
-   * Use [#build] to create the immutable [[Behavior]].
-   */
-  protected final class BehaviorBuilder(
+  // in order to keep all the constructors protected but still generate javadocs this class
+  // extends a public sealed trait where documentation is provided.
+  private[lagom] final class BehaviorBuilderImpl(
     state:       State,
     evtHandlers: Map[Class[_ <: Event], JFunction[_ <: Event, Behavior]],
     cmdHandlers: Map[Class[_ <: Command], JBiFunction[_ <: Command, CommandContext[Any], Persist[_ <: Event]]]
-  ) {
+  ) extends BehaviorBuilder {
 
     def this(state: State) = this(state, Map.empty, Map.empty)
```

Comment:
This constructor is no longer usable from `java` source code. 

It used to be usable because it was implemented in scala but that was a mistake (because some scala visibility rules are ignored from java code). [The method `PersistentEntity#newBehaviorBuilder() ` is documented](https://www.lagomframework.com/documentation/1.4.x/java/PersistentEntity.html#PersistentEntity-Stub) as the preferred way to obtain a `BehaviorBuilder`.